### PR TITLE
[Azure Load Testing] Onboard DataPlane SDK to Sync Stack

### PR DESCRIPTION
**Pull request by Harshan01**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/loadtesting/azure-developer-loadtesting/src/test/java/com/azure/developer/loadtesting/LoadTestAdministrationAsyncTests.java [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+
+    private BinaryData getFileBodyFromResource(String fileName) {
+        URL url = LoadTestAdministrationAsyncTests.class.getClassLoader().getResource(fileName);
+        return BinaryData.fromFile(new File(url.getPath()).toPath());
+    }
+
+    // Puts and Patches
+
+    @Test
+    @Order(1)
```
Comment (samvaity) [inline sdk/loadtesting/azure-developer-loadtesting/src/test/java/com/azure/developer/loadtesting/LoadTestAdministrationAsyncTests.java:63]: Is the order really required or they just don't have to be run parallel?

File: sdk/loadtesting/azure-developer-loadtesting/src/test/java/com/azure/developer/loadtesting/LoadTestAdministrationAsyncTests.java [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+    public void createOrUpdateTest() {
+        BinaryData body = BinaryData.fromObject(getTestBodyFromDict());
+        Mono<Response<BinaryData>> monoResponse = adminBuilder.buildAsyncClient().createOrUpdateTestWithResponse(newTestIdAsync, body, null);
```
Comment (samvaity) [inline sdk/loadtesting/azure-developer-loadtesting/src/test/java/com/azure/developer/loadtesting/LoadTestAdministrationAsyncTests.java:66]: Enable [AssertingClient Testing](https://github.com/Azure/azure-sdk-for-java/pull/33177/files#diff-53439b928744c1bab1221881a598ddc21919338f3ad896811f32c0213a207a56R69-R83), this makes sure sync call API's are running in completely synchronous call stack and vice-versa for async stack.

File: sdk/loadtesting/azure-developer-loadtesting/swagger/README.md [hunk 3]
```diff
@@ -11,6 +11,7 @@ security-scopes: https://cnt-prod.loadtesting.azure.com/.default
 artifact-id: azure-developer-loadtesting
 namespace: com.azure.developer.loadtesting
 generate-builder-per-client: true
+enable-sync-stack: true
```
Comment (samvaity) [inline sdk/loadtesting/azure-developer-loadtesting/swagger/README.md:14]: Include the autorest version used to generate this code.

File: sdk/loadtesting/azure-developer-loadtesting/src/test/java/com/azure/developer/loadtesting/LoadTestAdministrationAsyncTests.java (hunk 1 above)
Comment (Harshan01) [reply sdk/loadtesting/azure-developer-loadtesting/src/test/java/com/azure/developer/loadtesting/LoadTestAdministrationAsyncTests.java:63]: Yes the order is required. In record mode, we actually create a new test and do other operations on the created test. Then the last test is to delete the test

File: sdk/loadtesting/azure-developer-loadtesting/swagger/README.md (hunk 3 above)
Comment (Harshan01) [reply sdk/loadtesting/azure-developer-loadtesting/swagger/README.md:15]: Added

File: sdk/loadtesting/azure-developer-loadtesting/src/test/java/com/azure/developer/loadtesting/LoadTestAdministrationAsyncTests.java (hunk 2 above)
Comment (Harshan01) [reply sdk/loadtesting/azure-developer-loadtesting/src/test/java/com/azure/developer/loadtesting/LoadTestAdministrationAsyncTests.java]: Added AssertingClient

File: sdk/loadtesting/azure-developer-loadtesting/src/test/java/com/azure/developer/loadtesting/LoadTestingClientTestBase.java [hunk 4]
```diff
@@ -133,4 +120,93 @@ protected Map<String, Object> getServerMetricsBodyFromDict() {
...
+        String authorityHost = Configuration.getGlobalConfiguration().get("AUTHORITY_HOST");
+        if (authorityHost != null) {
+            credentialBuilder.authorityHost(authorityHost);
+        }
+        return credentialBuilder.build();
+    }
+
+    private HttpClient buildAsyncAssertingClient(HttpClient httpClient) {
+        return new AssertingHttpClientBuilder(httpClient)
+                .skipRequest((ignored1, ignored2) -> false)
```
Comment (samvaity) [inline sdk/loadtesting/azure-developer-loadtesting/src/test/java/com/azure/developer/loadtesting/LoadTestingClientTestBase.java:135]: can be removed.

File: sdk/loadtesting/azure-developer-loadtesting/src/test/java/com/azure/developer/loadtesting/LoadTestingClientTestBase.java [hunk 5]
```diff
@@ -133,4 +120,93 @@ protected Map<String, Object> getServerMetricsBodyFromDict() {
...
+                .assertAsync()
+                .build();
+    }
+
+    private HttpClient buildSyncAssertingClient(HttpClient httpClient) {
+        return new AssertingHttpClientBuilder(httpClient)
+                .skipRequest((ignored1, ignored2) -> false)
```
Comment (samvaity) [inline sdk/loadtesting/azure-developer-loadtesting/src/test/java/com/azure/developer/loadtesting/LoadTestingClientTestBase.java:142]: can be removed.

File: sdk/loadtesting/azure-developer-loadtesting/src/test/java/com/azure/developer/loadtesting/LoadTestingClientTestBase.java [hunk 6]
```diff
@@ -133,4 +120,93 @@ protected Map<String, Object> getServerMetricsBodyFromDict() {
...
+        }
+
+        LoadTestAdministrationClientBuilder loadTestAdministrationClientBuilder = new LoadTestAdministrationClientBuilder()
+                .endpoint(Configuration.getGlobalConfiguration().get("ENDPOINT", defaultEndpoint))
+                .httpClient(httpClient)
+                .httpLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC));
+
+        if (getTestMode() == TestMode.PLAYBACK) {
+            loadTestAdministrationClientBuilder
+                    .credential(request -> Mono.just(new AccessToken("this_is_a_token", OffsetDateTime.MAX)));
```
Comment (samvaity) [inline sdk/loadtesting/azure-developer-loadtesting/src/test/java/com/azure/developer/loadtesting/LoadTestingClientTestBase.java:173]:
```suggestion
                    .credential(new MockTokenCredential());
```

File: sdk/loadtesting/azure-developer-loadtesting/src/test/java/com/azure/developer/loadtesting/LoadTestingClientTestBase.java (hunk 5 above)
Comment (Harshan01) [reply sdk/loadtesting/azure-developer-loadtesting/src/test/java/com/azure/developer/loadtesting/LoadTestingClientTestBase.java]: Done

File: sdk/loadtesting/azure-developer-loadtesting/src/test/java/com/azure/developer/loadtesting/LoadTestingClientTestBase.java (hunk 4 above)
Comment (Harshan01) [reply sdk/loadtesting/azure-developer-loadtesting/src/test/java/com/azure/developer/loadtesting/LoadTestingClientTestBase.java]: Done

File: sdk/loadtesting/azure-developer-loadtesting/src/test/java/com/azure/developer/loadtesting/LoadTestingClientTestBase.java (hunk 6 above)
Comment (Harshan01) [reply sdk/loadtesting/azure-developer-loadtesting/src/test/java/com/azure/developer/loadtesting/LoadTestingClientTestBase.java]: Done

**Review by samvaity — APPROVED** [review]

**Pull request merged**